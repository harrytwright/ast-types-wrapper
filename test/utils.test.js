const chai = require('chai')
const recast = require('recast')
const { namedTypes: n, builders: b } = require('ast-types')

const assert = chai.assert

const { identifier } = require('../src/utils')

describe('utils.identifier', function () {
  it('should create an identifier', function () {
    const name = identifier('name')
    assert(n.Identifier.check(name), 'name is Identifier')
  })

  it('should use a passed identifier', function () {
    const builtName = b.identifier('name')
    const name = identifier(builtName)

    assert(builtName === name, 'name is same as builtName')
  })

  it('should throw an error', function () {
    assert.throw(() => identifier(5), Error, 'Invalid identifier type: ' + 5)
  })

  it('should prints properly', function () {
    const code = recast.print(identifier('name')).code
    assert(code === 'name', 'Should turn name into an identifier')
  })
})
