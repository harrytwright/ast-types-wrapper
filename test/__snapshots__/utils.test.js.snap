// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils.identifier should create an identifier 1`] = `
Object {
  "comments": null,
  "loc": null,
  "name": "name",
  "optional": false,
  "type": "Identifier",
  "typeAnnotation": null,
}
`;
